frontend/packages/helm-plugin/src/components/forms/ProjectHelmChartRepository/CreateProjectHelmChartRepositoryForm.tsx

```diff
@@ -1,121 +1,110 @@
 import * as React from 'react';
-import { TextInputTypes } from '@patternfly/react-core';
 import { FormikProps, FormikValues } from 'formik';
-import * as fuzzy from 'fuzzysearch';
 import * as _ from 'lodash';
 import { useTranslation } from 'react-i18next';
-import FormSection from '@console/dev-console/src/components/import/section/FormSection';
-import { ExpandCollapse, history } from '@console/internal/components/utils';
-import { ConfigMapModel, SecretModel } from '@console/internal/models';
 import {
-  InputField,
   FormFooter,
   FormBody,
   FormHeader,
   FlexForm,
-  ResourceDropdownField,
-  CheckboxField,
-} from '@console/shared';
+  SyncedEditorField,
+  YAMLEditorField,
+} from '@console/shared/src';
+import { downloadYaml } from '@console/shared/src/components/editor/yaml-download-utils';
+import { EditorType } from '@console/shared/src/components/synced-editor/editor-toggle';
+import { safeJSToYAML } from '@console/shared/src/utils/yaml';
+import { ProjectHelmChartRepositoryModel } from '../../../models';
+import { ProjectHelmChartRepositoryType } from '../../../types/helm-types';
+import CreateProjectHelmChartRepositoryFormEditor from './CreateProjectHelmChartRepositoryFormEditor';
+import {
+  convertToForm,
+  convertToProjectHelmChartRepository,
+} from './projecthelmchartrepository-create-utils';
 
-const CreateProjectHelmChartRepositoryForm: React.FC<FormikProps<FormikValues>> = ({
+const CreateProjectHelmChartRepositoryForm: React.FC<FormikProps<FormikValues> & {
+  resource: ProjectHelmChartRepositoryType;
+  handleCancel: () => void;
+}> = ({
+  resource,
   errors,
   handleSubmit,
+  setFieldTouched,
+  handleCancel,
   status,
   isSubmitting,
   dirty,
+  values: { editorType, formData, yamlData },
 }) => {
   const { t } = useTranslation();
-  const autocompleteFilter = (strText, item): boolean => fuzzy(strText, item?.props?.name);
+
+  const LAST_VIEWED_EDITOR_TYPE_USERSETTING_KEY =
+    'helm-plugin.createProjectHelmChartRepository.editor.lastView';
+
+  React.useEffect(() => {
+    if (formData.name || formData.repoUrl) {
+      setFieldTouched('formData.repoName', true);
+      setFieldTouched('formData.repoUrl', true);
+    }
+  }, [setFieldTouched, formData.repoName, formData.repoUrl]);
```

Comment:
This will fire on every keystroke, can we move this setFieldTouch to `sanitizeToForm` function, so when you switch back to form and if the value is set only then this call will be fired.

Reply:
I have changed it
Added it inside the `sanitizeToForm` function.